components/collator/src/options.rs

```diff
@@ -240,6 +240,47 @@ pub enum MaxVariable {
     Currency = 3,
 }
 
+/// Whether to distinguish case in sorting, even for sorting levels higher
+/// than tertiary, without having to use tertiary level just to enable case level differences.
+#[derive(Eq, PartialEq, Debug, Copy, Clone)]
+#[repr(u8)]
+#[non_exhaustive]
+pub enum CaseLevel {
```

Comment:
Why not use simple booleans for the purely on/off settings?

Reply:
summarizing offline discussion that also included @sffc / @Manishearth : Rust idiomatic code prefers enums instead of booleans because they are more extensible and more readable (ex: `Numeric::On` / `Numeric::Off` versus `true` / `false`).  And maybe we can't predict when we choose a boolean and find out later that it doesn't represent the space of values, ex: number formatter in ICU once only used a boolean for the grouping separator.